src/meta/src/hummock/manager/mod.rs

```diff
@@ -660,29 +659,62 @@ where
             compaction_group_id,
         );
 
-        #[cfg(test)]
-        {
-            drop(compaction_guard);
-            self.check_state_consistency().await;
-        }
-
         ret
     }
 
+    #[named]
+    pub async fn cancel_compact_task(&self, compact_task: &CompactTask) -> Result<bool> {
+        let mut compaction_guard = write_lock!(self, compaction).await;
+        let compaction = compaction_guard.deref_mut();
+        self.report_compact_task_impl(None, compact_task, compaction)
+            .await
+    }
+
+    #[named]
     pub async fn get_compact_task(
         &self,
         compaction_group_id: CompactionGroupId,
     ) -> Result<Option<CompactTask>> {
-        self.get_compact_task_impl(compaction_group_id, None).await
+        let mut compaction_guard = write_lock!(self, compaction).await;
```

Comment:
There is a doubt about this, and its the reason why unit-test fails

we report_compact_task if the task that is a `trivial_move_task`, and return `None` to caller which may be a bit misunderstood.

Because its not that there is no task, but the task do not need to be sent to the compactor for execution. Therefore, if i understand corrently, can we report it after get_compact_task (check the task_status if `true`), or are there other considerations here?